projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts

```diff
@@ -71,6 +95,42 @@ export class NavigationUIComponent implements OnDestroy {
     );
   }
 
+  /** During initialization of this component, we will check the resetMenuOnClose flag and attach a menu reset listener if needed.
+   */
+  ngOnInit() {
+    if (this.resetMenuOnClose) this.resetOnMenuCollapse();
+  }
+
+  /**
+   * This method performs the action of resetting the menu (close all sub menus and return to main options)
+   * when the menu is closed.
+   */
+  resetOnMenuCollapse(): void {
+    this.subscriptions.add(
+      this.hamburgerMenuService.isExpanded
+        .pipe(distinctUntilChanged())
+        .subscribe((isExpanded: boolean) => {
+          //only react when expanded and proper number of nav node are present.
+          if (isExpanded && this.node.children?.length > 6) {
```

Comment:
Where does the number `6` come from? Better to set a constant and reference that so we can see what these numbers refer to (eg. this.node.children?.length > NODE_LIMIT, note: `NODE_LIMIT` is just an example var, I still don't know what it refers to).